### PR TITLE
test: make chat api tests fully independent

### PR DESCRIPTION
**Pull request by MarcosSpessatto**
https://rocketchat.atlassian.net/browse/CORE-150

### CODE REVIEW

File: apps/meteor/tests/end-to-end/api/05-chat.js [hunk 1]
```diff
@@ -2167,13 +2169,21 @@ describe('[Chat]', function () {
...
+			).body.discussion;
 		});
 
+		after(() =>
+			Promise.all([
+				deleteRoom({ type: 'c', roomId: testChannel._id }),
+				request
+					.post(api('rooms.delete'))
+					.set(credentials)
+					.send({ roomId: testChannel._id })
```
Comment (sampaiodiego) [inline apps/meteor/tests/end-to-end/api/05-chat.js:2181]: I guess this room is being deleted twice?

File: apps/meteor/tests/end-to-end/api/05-chat.js [hunk 2]
```diff
@@ -785,44 +791,40 @@ describe('[Chat]', function () {
...
-						.get(api('chat.getMessage'))
-						.set(credentials)
-						.query({
-							msgId: ytEmbedMsgId,
-						})
-						.expect('Content-Type', 'application/json')
-						.expect(200)
-						.expect((res) => {
-							expect(res.body).to.have.property('message').to.have.property('urls').to.be.an('array').that.is.not.empty;
+				request
```
Comment (sampaiodiego) [inline apps/meteor/tests/end-to-end/api/05-chat.js:794]: not sure about removing the `setTimeout` here.. the oembed action runs in a `afterSaveMessage` callback and it depends on how fast the CI can make the request and also how fast YT can respond to the request.. 

I have to admit though I'm shocked the test didn't fail.. but maybe removing the timeout can make it flaky

File: apps/meteor/tests/end-to-end/api/05-chat.js [hunk 3]
```diff
@@ -785,44 +791,40 @@ describe('[Chat]', function () {
...
-						.get(api('chat.getMessage'))
-						.set(credentials)
-						.query({
-							msgId: imgUrlMsgId,
-						})
-						.expect('Content-Type', 'application/json')
-						.expect(200)
-						.expect((res) => {
-							expect(res.body).to.have.property('message').to.have.property('urls').to.be.an('array').that.is.not.empty;
+				request
```
Comment (sampaiodiego) [inline apps/meteor/tests/end-to-end/api/05-chat.js:814]: same concern as above.. 

I wonder though if we should change both these end-to-end tests for some unit tests.. what we need is to check if the oembed code is doing it's thing.. relying on external services (like on the youtube link above) is prone to fail.. wdyt?

**Review by sampaiodiego — CHANGES_REQUESTED** [review]

File: apps/meteor/tests/end-to-end/api/05-chat.js (hunk 3 above)
Comment (MarcosSpessatto) [reply apps/meteor/tests/end-to-end/api/05-chat.js:814]: Agreed, I'll remove (skip) these tests in this PR and I will create that in another PR, wdyt?

**Review by sampaiodiego — DISMISSED** [review]

**Review by sampaiodiego — APPROVED** [review]

**Pull request merged**